lib/mongoid/factory.rb

```diff
@@ -60,14 +60,16 @@ def build(klass, attributes = nil)
     # @param [ Hash ] selected_fields Fields which were retrieved via
     #   #only. If selected_fields are specified, fields not listed in it
     #   will not be accessible in the returned document.
+    # @param [ true | false ] execute_callbacks Flag specifies whether find
+    #   callbacks should be run.
     #
     # @return [ Document ] The instantiated document.
-    def from_db(klass, attributes = nil, criteria = nil, selected_fields = nil)
+    def from_db(klass, attributes = nil, criteria = nil, selected_fields = nil, execute_callbacks: true)
```

Comment:
Think the code would be clearer if this variable were called `defer_callbacks`.

To me, if `execute_callbacks: false` it implies that the callbacks will never be executed; its not obvious that they will be deferred until later.

Reply:
Good point, I was still workshopping names as this is a draft PR but thanks for the idea